### PR TITLE
remove `InternalDebugger.track_load_dataloader_call`

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Part of #9661 

Removes the dataloader tracking logic from the dev debugger. Tests are updated to cover what the "debugger" offered for tracking the calls inside the trainer.
This is the last step before we can remove the InternalDebugger completely.

Note the elegancy of using mocks in the newly written tests.

Next PR: #9680

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

### CODE REVIEW

File: tests/trainer/test_dataloaders.py [hunk 1]
```diff
@@ -1123,63 +1121,65 @@ def test_dataloaders_load_only_once_val_interval(tmpdir):
...
         max_epochs=3,
     )
-    trainer.fit(model)
-    assert trainer.state.finished, f"Training failed with {trainer.state}"
 
-    trainer.test()
+    tracker = Mock()
+    model.train_dataloader = Mock(wraps=model.train_dataloader)
+    model.val_dataloader = Mock(wraps=model.val_dataloader)
+    model.test_dataloader = Mock(wraps=model.test_dataloader)
```
Comment (awaelchli) [inline tests/trainer/test_dataloaders.py:1128]: reviewer: witness the elegancy of testing with these mocks

**Review by carmocca — APPROVED** [review]
TIL about `attach_mock` 👏

**Review by ananthsub — APPROVED** [review]
very nice!

File: tests/trainer/test_dataloaders.py [hunk 2]
```diff
@@ -1123,63 +1121,65 @@ def test_dataloaders_load_only_once_val_interval(tmpdir):
...
 
-    assert len(trainer.dev_debugger.val_dataloader_calls) == 10
-    assert len(trainer.dev_debugger.test_dataloader_calls) == 1
-    assert len(trainer.dev_debugger.train_dataloader_calls) == 3
+    tracker.attach_mock(model.train_dataloader, "train_dataloader")
+    tracker.attach_mock(model.val_dataloader, "val_dataloader")
+    tracker.attach_mock(model.test_dataloader, "test_dataloader")
```
Comment (rohitgr7) [inline tests/trainer/test_dataloaders.py:1132]: nice! TIL

**Review by rohitgr7 — APPROVED** [review]
